src/maxminddb.c

```diff
@@ -795,6 +798,8 @@ static int populate_description_metadata(MMDB_s *mmdb,
             " status = %d (%s)",
             status,
             MMDB_strerror(status));
+        if (MMDB_OUT_OF_MEMORY_ERROR != status)
+            MMDB_free_entry_data_list(member);
```

Comment:
If we make `MMDB_get_entry_data_list()` clean up after itself, this shouldn't be needed.

Reply:
https://github.com/maxmind/libmaxminddb/pull/356#discussion_r1833681864